core/View.php

```diff
@@ -316,12 +317,16 @@ protected function renderTwigTemplate()
     {
         try {
             $output = $this->twig->render($this->getTemplateFile(), $this->getTemplateVars());
-        } catch (Exception $ex) {
+        } catch (Error $ex) {
             // twig does not rethrow exceptions, it wraps them so we log the cause if we can find it
-            $cause = $ex->getPrevious();
-            Log::debug($cause === null ? $ex : $cause);
+            //TODO: ugly hack to see full twig exception (with path to template)
+            $context = $ex->getSourceContext();
+            $message=$ex->getMessage()."<br>";
+            $message .= $context->getName() . "<br>";
+            $message .= $context->getPath() .":".$ex->getLine(). "<br>";
+            Log::debug($message);
+            return $message;
```

Comment:
It would be great if someone who knows more about Matomo logging could replace this with "the proper way to do it".

Reply:
It's just logging via `StaticContainer::get(LoggerInterface::class)->debug(...)`.

Reply:
Actually I wonder if that is needed at all. I've removed the try catch locally and put an parser error in _logo.twig, which results in this exception being logged:

```
ERROR Piwik\ExceptionHandler[2020-05-25 07:47:04 UTC] [4102e] Uncaught exception: /srv/matomo/plugins/CoreHome/templates/_logo.twig(7): Unexpected token "name" of value "Logo" ("end of statement block" expected).
ERROR Piwik\ExceptionHandler[2020-05-25 07:47:04 UTC] [4102e] Unexpected token "name" of value "Logo" ("end of statement block" expected).
ERROR Piwik\ExceptionHandler[2020-05-25 07:47:04 UTC] [4102e] #0 /srv/matomo/vendor/twig/twig/src/TokenParser/IfTokenParser.php(38): Twig\TokenStream->expect(3)
```

But the error screen actually only shows the message `Unexpected token "name" of value "Logo" ("end of statement block" expected)`

Reply:
removed that now, as imho the default handling is fine enough with twig 3